pkg/engine/template_generator.go

```diff
@@ -205,6 +206,29 @@ func (t *TemplateGenerator) getMasterCustomData(cs *api.ContainerService, textFi
 // getTemplateFuncMap returns all functions used in template generation
 func (t *TemplateGenerator) getTemplateFuncMap(cs *api.ContainerService) template.FuncMap {
 	return template.FuncMap{
+		"IsAzureStackCloud": func() bool {
+			var cloudProfileName string
+			if cs.Properties.CustomCloudProfile != nil {
+				if cs.Properties.CustomCloudProfile.Enviornment != nil {
+					cloudProfileName = cs.Properties.CustomCloudProfile.Enviornment.Name
+				}
+			}
+			return strings.EqualFold(cloudProfileName, api.AzureStackCloud)
```

Comment:
Just curious why the `EqualFold` usage here and not a strict equality check?

Reply:
in the environment.go, 
	if strings.EqualFold(name, "AZURESTACKCLOUD") {
		return EnvironmentFromFile(os.Getenv(EnvironmentFilepathName))
	}
it load the cloud name case insensitive, so I followed the pattern.